Xamarin.PropertyEditing.Mac/Controls/Custom/MaterialView.cs

```diff
@@ -139,55 +147,61 @@ MaterialColorLayer CreateLayer (CommonColor color)
 			y += height + 6;
 			x = 0;
 
-			var accent = new CALayer {
-				CornerRadius = 3,
-				MasksToBounds = true,
-				Frame = new CGRect (x, y, Frame.Width, height),
-				BorderColor = new CGColor (.5f, .5f, .5f, .5f),
-				BorderWidth = 1
-			};
-			Layer.AddSublayer (accent);
-			width = Frame.Width / ViewModel.AccentColorScale.Count ();
+			width = FrameWidth / ViewModel.AccentColorScale.Count ();
 			foreach (var color in ViewModel.AccentColorScale) {
 				var l = CreateLayer (color.Value);
 				l.ColorType = MaterialColorType.Accent;
 				l.IsSelected = color.Value == ViewModel.AccentColor || color.Value == ViewModel.Color;
-				l.Frame = new CGRect (x, 0, width, height);
-				AddToolTip (new CGRect (x, y, width, height), new NSString (color.ToString ()), IntPtr.Zero);
-				accent.AddSublayer (l);
+				l.Frame = new CGRect (0, 0, width, height);
+
+				var accentColourButton = new FocusableButton {
+					Frame = new CGRect (x, y, width, height),
+					WantsLayer = true,
+					Layer = l,
+					ToolTip = color.ToString (),
+					Transparent = false,
+					TranslatesAutoresizingMaskIntoConstraints = true,
+				};
+
+				accentColourButton.Activated += MaterialColourButton_Activated;
+
+				AddSubview (accentColourButton);
+
 				x += width;
 			}
+
+			Window?.RecalculateKeyViewLoop ();
```

Comment:
`Window` is null here the first time this is called. See the commit I pushed to call it earlier

Reply:
Thanks!